src/jit.rs

```diff
@@ -1758,9 +1797,10 @@ mod tests {
                     INSTRUCTION_COUNT
                 };
                 let machine_code_length_per_instruction =
-                    (machine_code_length as f64 / instruction_count as f64 + 0.5) as usize;
+                    machine_code_length as f64 / instruction_count as f64;
                 assert!(
-                    machine_code_length_per_instruction <= MAX_MACHINE_CODE_LENGTH_PER_INSTRUCTION
+                    (machine_code_length_per_instruction + 0.5) as usize
```

Comment:
You don't need the floating point for the division. `machine_code_length % instruction_count` is always zero. In addition, our overestimation of `MAX_MACHINE_CODE_LENGTH_PER_INSTRUCTION` already covers the extra 0.5 , so we can remove it.